### DISCUSSION TITLE
圖片OCR轉文字 驗證碼多了空白

### DISCUSSION THREAD

**Discussion by shushilin1** [hiroi-sora/Umi-OCR#511] [General]
如附檔：

圖片每個文字間隔不依  , 較寬的地方會被誤認為空白. 請問有選項可以解決？？　　謝謝
![test11](https://github.com/hiroi-sora/Umi-OCR/assets/169340110/f4742fdd-aebd-4087-8459-8cd81859a2e7)

**Comment by hiroi-sora** [comment]
你好，这个需求可以简单地修改一下代码来实现。请参考以下步骤：

1. 用文本编辑器，如记事本，打开 `UmiOCR-data\py_src\ocr\tbpu\parser_single_none.py`
2. 将 `def run(self, textBlocks):` 那一坨函数，改为如下：
```python
    def run(self, textBlocks):
        textBlocks = super().run(textBlocks)
        for i in range(len(textBlocks)):
            textBlocks[i]["text"] = textBlocks[i]["text"].replace(" ", "") # 去除句中空格
            textBlocks[i]["end"] = "" # 去除句尾换行
        return textBlocks
```
（请保持格式复制代码，不要丢掉缩进）

3. 保存后打开Umi，排版解析方案选择 **单栏-无换行**

即可。
![image](https://github.com/hiroi-sora/Umi-OCR/assets/56373419/c3933cb9-72f5-46e3-b1dd-0442d4423910)

**Comment by shushilin1** [comment]
谢谢~~ 免费的比花钱买的还给力
可惜没打赏功能

**Reply by shushilin1** [reply]
大大请帮我看一下，　空格还是存在. 感恩
![back1](https://github.com/hiroi-sora/Umi-OCR/assets/169340110/5c7c82d4-47a4-41c9-a114-78517bc18e00)
![back1-2](https://github.com/hiroi-sora/Umi-OCR/assets/169340110/cd274fcd-c104-4869-b70f-5ff7cc385d5d)
![back2](https://github.com/hiroi-sora/Umi-OCR/assets/169340110/fb519b37-a589-4f8c-ae64-e5246623f170)

**Comment by hiroi-sora** [comment]
哦，抱歉，请再改一下：
```python
        for i in range(len(textBlocks) - 1):
```
这一行，将`-1`删掉，改为：
```python
        for i in range(len(textBlocks)):
```

**Reply by shushilin1** [reply]
謝謝大大,可以了
![22](https://github.com/hiroi-sora/Umi-OCR/assets/169340110/6c7a85e0-dd43-4e2b-b4f0-1961b0c8e84c)
